### PR TITLE
Feature/2587

### PR DESCRIPTION
**Pull request by jablpiotrek**
### Related issues

https://github.com/DivanteLtd/vue-storefront/issues/2587

### Short description and why it's useful

It adds "Clear cart" button which... clears cart!

### Screenshots of visual changes before/after (if there are any)

Zrzut ekranu 2019-03-21 14 21 08

### CODE REVIEW

File: src/themes/default/components/core/blocks/Microcart/Microcart.vue [hunk 1]
```diff
@@ -184,6 +189,20 @@ export default {
     },
     onEscapePress () {
       this.closeMicrocart()
+    },
+    clearCart () {
+      this.$store.dispatch('notification/spawnNotification', {
+        type: 'warning',
+        item: this.product,
+        message: i18n.t('Are you sure you would like to remove all the itmes from the shopping cart?'),
```
Comment (Aekal) [inline src/themes/default/components/core/blocks/Microcart/Microcart.vue]: @jablpiotrek typo 'itmes' :D

Comment (jablpiotrek) [reply src/themes/default/components/core/blocks/Microcart/Microcart.vue]: https://media.giphy.com/media/12soWQ91xpTDVu/giphy.gif

Comment (jablpiotrek) [reply src/themes/default/components/core/blocks/Microcart/Microcart.vue]: done ")

File: CHANGELOG.md [hunk 2]
```diff
@@ -7,10 +7,10 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 ## [1.10.0-rc.1] - UNRELEASED
 
 ### Added
-- 
+- "Clear cart" button in the cart
```
Comment (patzick) [inline CHANGELOG.md]:
```suggestion
- "Clear cart" button in the cart - @jablpiotrek (#2587)
```

File: src/themes/default/components/core/blocks/Microcart/Microcart.vue [hunk 3]
```diff
@@ -184,6 +189,20 @@ export default {
     },
     onEscapePress () {
       this.closeMicrocart()
+    },
+    clearCart () {
+      this.$store.dispatch('notification/spawnNotification', {
+        type: 'warning',
+        item: this.product,
```
Comment (patzick) [inline src/themes/default/components/core/blocks/Microcart/Microcart.vue]:
```suggestion
```
i think this shouldn't be here

**Comment by daksamit** [comment]
@jablpiotrek There is a notification typo - 
image
'itmes' instead of 'items'

**Comment by patzick** [comment]
@daksamit in code there is no typo:)

File: CHANGELOG.md (hunk 2 above)
Comment (jablpiotrek) [reply CHANGELOG.md]: Done, thanks :)

File: src/themes/default/components/core/blocks/Microcart/Microcart.vue (hunk 3 above)
Comment (jablpiotrek) [reply src/themes/default/components/core/blocks/Microcart/Microcart.vue]: I thought I removed it 😿 
Thanks for spotting.

**Comment by jablpiotrek** [comment]
All comments included in change.

**Comment by jablpiotrek** [comment]
@filrak can we merge it, or maybe you have some further comments?

**Comment by pkarw** [comment]
@patzick dont we need just a spacing between buttons? Can we merge this one in today?

**Comment by ArturDivante** [comment]
Button works correctly, but from UX perspective it's location in a bit unfortunate. IMHO bare minimum should be to separate it from 'Instant checkout' button. But in long term perspective, best place for this button would be in the upper part of MicroCart window. Currently user with any products in the cart (even two, depending on screen parameters) needs to scroll down microcart even to realize that this button exist. In most cases users will just manually remove prducts rather than scrolli down to find 'Clear Cart' button below checkout options.

clearCart

**Pull request merged**